src/transformers/models/blenderbot/tokenization_blenderbot.py

```diff
@@ -413,19 +410,14 @@ def build_inputs_with_special_tokens(self, token_ids_0: List[int], token_ids_1:
         """
         return token_ids_0 + [self.eos_token_id]
 
-    def _build_conversation_input_ids(self, conversation: "Conversation") -> List[int]:
-        inputs = []
-        for is_user, text in conversation.iter_texts():
-            if is_user:
-                # We need to space prefix as it's being done within blenderbot
-                inputs.append(" " + text)
-            else:
-                # Generated responses should contain them already.
-                inputs.append(text)
-
-        full_string = "  ".join(inputs)
-        input_ids = self.encode(full_string)
-        if len(input_ids) > self.model_max_length:
-            input_ids = input_ids[-self.model_max_length :]
-            logger.warning(f"Trimmed input from conversation as it was longer than {self.model_max_length} tokens.")
-        return input_ids
+    @property
+    def default_chat_template(self):
+        # BlenderBot actually tokenizes messages together with a "  " separator between them, so we try to imitate that
+        return (
+            "{% for message in messages %}"
+            "{% if message.role == 'user' %}{{ ' ' }}{% endif %}"
+            "{{ message.content }}"
+            "{% if not loop.last %}{{ '  ' }}{% endif %}"
+            "{% endfor %}"
+            "{{ eos_token }}"
+        )
```

Comment:
Love it! Much cleaner this way IMO

Reply:
It seems like there is still a `_build_conversation_input_ids` in the `tokenization_bloom_fast.py` file

Reply:
Would be nice to add a dummy example of the format this will generate

Reply:
Cleaned up all the `_build_conversation_input_ids` methods left over and added docstrings to all the `default_chat_template` properties!